modules.d

```diff
@@ -48,18 +48,16 @@ string readBuiltin(string[] paths)(int index){
 	}
 }
 
-Scope getPreludeScope(ErrorHandler err, Location loc){
+Scope getPreludeScope(ErrorHandler err){
 	import ast.semantic_: semantic;
 	if(preludeScope) return preludeScope;
 	preludeScope = new TopScope(err);
 	preludeSrc = new Source(".prelude", readBuiltin!preludePaths(preludeIndex));
 	int nerr = err.nerrors;
 	auto exprs = parseSource(preludeSrc, err);
 	exprs = semantic(exprs, preludeScope);
-	if(nerr != err.nerrors) {
-		if(loc.line) preludeScope.error("failed to import prelude", loc);
-		else preludeScope.message("failed to import prelude");
-	}
+	// failure to import prelude is non-recoverable.
+	assert(nerr == err.nerrors);
```

Comment:
What is the motivation for this commit? The prelude not loading is not an internal logic error in the program, so an assertion is not suitable for this check. I pulled the other commits.